gdp/small_batch/gdp_small_batch.py

```diff
@@ -124,64 +134,78 @@ def Y_exists_or_not(m, k, j):
         for j in m.j:
             m.Y[k, j].associate_binary_var(m.Y_exists[k, j].indicator_var)
 
-    # ____________________________
-
-    # Objective
+    # Objective function to minimize cost
     def obj_rule(m):
         return sum(m.alpha[j]*(pe.exp(m.n[j] + m.beta[j]*m.v[j])) for j in m.j)
 
+    # Define the objective
     m.obj = pe.Objective(rule=obj_rule, sense=pe.minimize)
 
     return m
 
 
 def external_ref(m, x, logic_expr=None):
+    # Create a dictionary to hold external reference for each stage
     ext_var = {}
     p = 0
     for j in m.j:
+        # Assign each stage j to its corresponding external reference from x
         ext_var[j] = x[p]
         p = p+1
 
+    # Go through each unit k and stage j
     for k in m.k:
         for j in m.j:
+            # If unit k is the selected unit for stage j according to ext_var
             if k == ext_var[j]:
+                # Fix the boolean variable Y[k,j] to True, meaning unit k is chosen for stage j
                 m.Y[k, j].fix(True)
-                m.Y_exists[k, j].indicator_var.fix(
-                    True)  # IS THIS REQUIRED????
-                m.Y_not_exists[k, j].indicator_var.fix(
-                    False)  # IS THIS REQUIRED????
+                # Fix the indicator variable of Y_exists[k, j] to True, because unit k exists
+                m.Y_exists[k, j].indicator_var.fix(True)  # IS THIS REQUIRED????
```

Comment:
Add a todo as follows:
# TODO: Is this required?

Reply:
This will be addressed in a separate PR